source/dubregistry/dbcontroller.d

```diff
@@ -325,14 +330,22 @@ class DbController {
 				.array;
 		}
 
-		return m_packages
-			.find(["$text": ["$search": query]], ["score": bson(["$meta": "textScore"])])
-			.sort(["score": bson(["$meta": "textScore"])])
-			.limit(50) // fix #341 - avoid hitting mem-limit of MongoDB's sort
+		auto pkgs = m_packages
+			.find(["$text": ["$search": query]], ["textScore": bson(["$meta": "textScore"])])
+			.sort(["textScore": bson(["$meta": "textScore"])]) // sort to only keep most relevant results
+			.limit(50) // limit irrelevant sort results (fixes #341)
 			.map!(deserializeBson!DbPackage)
-			.array
-			// sort by bucketized score preserving FTS score order
-			.sort!((a, b) => a.stats.score.round > b.stats.score.round, SwapStrategy.stable)
+			.array;
+
+		// normalize textScore to same scale as package score
+		immutable minMaxTS = pkgs.map!(p => p.textScore).fold!(min, max)(0.0f, 0.0f);
+		immutable scale = (DbPackageStats.maxScore - DbPackageStats.minScore) / (minMaxTS[1] - minMaxTS[0]);
+		foreach (ref pkg; pkgs)
+			pkg.textScore = (pkg.textScore - minMaxTS[0]) * scale + DbPackageStats.minScore;
+
+		// sort found packages by weighted textScore and package score
+		return pkgs
+			.sort!((a, b) => a.stats.score + 2 * a.textScore > b.stats.score + 2 * b.textScore)
```

Comment:
This is a bit weird as ideally this should be handled by the database, but I'm guessing MongoDB doesn't support this :/